src/server.c

```diff
@@ -2860,6 +2860,11 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
     /* Close clients that need to be closed asynchronous */
     freeClientsInAsyncFreeQueue();
 
+    /* Incrementally trim replication backlog, 10 times the normal speed is
+     * to free replication backlog as much as possible. */
+    if (server.repl_backlog)
+        incrementalTrimReplicationBacklog(10*REPL_BACKLOG_TRIM_BLOCKS_PER_CALL);
+
```

Comment:
i feel that replicationCron could be too slow / late, and maybe introduce excessive eviction or client disconnection in that time, and that can be also said on serverCron. doing smaller steps in beforeSleep sounds better to me.

but maybe we should do some benchmark or analyze the worse case, instead of judge by feelings?

Reply:
in replicationCron, it is too slow. Initially i put it in replicationCron, it costed dozens of seconds for trimming replication backlog in my tcl tests. we free 1M buffer blocks per call, in beforeSleep, we free 10M

Reply:
I see. Thanks for explaining. I don't feel strongly either way as long as we document the reason why we put it here (e.g. for speed).